rust/src/jsonbuilder.rs

```diff
@@ -502,6 +502,20 @@ impl JsonBuilder {
         Ok(self)
     }
 
+    /// Set a key and a string value on an object, with a limited size
+    pub fn set_string_limited(
+        &mut self, key: &str, val: &str, limit: usize,
+    ) -> Result<&mut Self, JsonError> {
+        if val.len() > limit {
+            let valtrunc = &val[..limit];
+            let outstr = format!("{valtrunc}[truncated]");
```

Comment:
would it make sense to have `somevalu[truncated 99 additional bytes]` or something similar?

Reply:
Sounds good, would also make sense for `set_string_from_bytes_limited()`.

Reply:
Implemented this; will be in next PR.